File: codechef/snackdown2016/kttable.py
import sys

lines = 0
tc = 0
AA = []
N = 0
A = []
B = []

for line in sys.stdin:
    if lines == 0:
        tc = line

    if lines % 3 == 1:
        N = int(line)

    elif lines % 3 == 2:
        A = [int(a) for a in line.strip("\n").split(" ")]
        for n in range(0, N):
            if n == 0:
                AA.append(A[n])
            else:
                AA.append(A[n] - A[n-1])

    elif lines > 0 & lines % 3 == 0:
        B = [int(b) for b in line.strip("\n").split(" ")]
        count = 0
        for n in range(0, N):
            if B[n] <= AA[n]:
                count += 1

        print count

    lines += 1

File: codechef/snackdown2016/kttable.js
const readline = require('readline');
const fs = require('fs');

var T;

process.stdin.resume();
process.stdin.setEncoding('utf8');

const rl = readline.createInterface({
    input: process.stdin,
    terminal: false
});

var lines = 0;
var tc = {};

rl.on('line', function(data) {
    if (lines === 0) {
        T = parseInt(data);
    }

    if (lines % 3 === 1) {
        tc.N = parseInt(data);
    }

    if (lines % 3 === 2) {
        tc.A = data.split(' ');
        tc.A = tc.A.map(function(v, i) {
            if (i === 0)
                return v;
            else
                return v - tc.A[i - 1];
        });
    }

    if (lines > 0 && lines % 3 === 0) {
        tc.B = data.split(' ');
        executeTest(tc);
        tc = {};
    }

    lines++;
});

function executeTest(tc) {
    var count = 0;
    for (var i = 0; i < tc.B.length; i++) {
        if (parseInt(tc.B[i]) <= tc.A[i])
            count++
    }

    console.log(count);
}